src/main/java/seedu/duke/parser/Parser.java

```diff
@@ -1,11 +1,422 @@
+            command = new DeleteTaskCommand();
+            break;
+        default:
+            throw new UnknownCommandException();
+        }
+        return command;
+    }
+
+    /**
+     * Parses user input to determine in-module command specified.
+     *
+     * @param input full user input string
+     * @return an integer representing the command specified
+     */
+    private int parseInModuleCommandsFromInput(String input) {
+        if (input.equalsIgnoreCase("help")) {
+            return PRINT_HELP_MODULE;
+        } else if (input.equalsIgnoreCase("close")) {
+            return EXIT_MODULE;
+        } else if (input.equalsIgnoreCase("info")) {
+            return LIST_MODULE_INFO;
+        } else if (input.equalsIgnoreCase("lessons")) {
+            return LIST_LESSONS;
+        } else if (input.equalsIgnoreCase("link")) {
+            return OPEN_LINK;
+        } else if (input.equalsIgnoreCase("tasks")) {
+            return LIST_TASKS;
+        } else if (input.equalsIgnoreCase("mark")) {
+            return MARK_DONE;
+        } else if (input.equalsIgnoreCase("unmark")) {
+            return UNMARK_DONE;
+        } else if (input.equalsIgnoreCase("teacher")) {
+            return LIST_TEACHING_STAFF;
+        } else if (startsWith(input, "add lesson")) {
+            return ADD_LESSON;
+        } else if (startsWith(input, "delete lesson")) {
+            return DEL_LESSON;
+        } else if (startsWith(input, "add task")) {
+            return ADD_TASK;
+        } else if (startsWith(input, "delete task")) {
+            return DEL_TASK;
+        } else {
+            return UNKNOWN_COMMAND;
+        }
+    }
+
+    /**
+     * Parses details of new lesson from user input string.
+     *
+     * @param input full user input string
+     * @return a Lesson object with the details entered by the user. Fields not found will be left as an empty string.
+     */
+    private Lesson parseNewLessonDetails(String input) {
+        // TODO - validate input
+
+        // initialize an array of empty strings to store lesson details
+        String[] allDetails = new String[LESSON_MAX_DETAILS];
+        Arrays.fill(allDetails, "");
+
+        // to remove only the first two words "add lesson"
+        String[] lessonDetails = input.trim().split(" ", 3);
+
+        // split the details field using DELIMITER to get the individual detail fields
+        String[] details = lessonDetails[2].split(DELIM);
+
+        // store detail fields that have been filled by the user into an array
+        // if user did not enter that field, it will remain as an empty string
+        for (int i = 0; i < details.length && i < allDetails.length; i++) {
+            allDetails[i] = details[i].trim();
+        }
+
+        // Creating Lesson Object
+        String type = allDetails[LESSON_TYPE_INDEX].toUpperCase();
+        // throws "illegal argument exception" if enum value is wrong
+        LessonType TYPE = LessonType.valueOf(type);
```

Comment:
What happens after the exception is thrown?

Reply:
For now we are coding for the happy case so it's irrelevant. I wrote that comment to remind us later on to catch that exception when user enters an invalid lesson type